spec/services/invoices/finalize_open_credit_service_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Invoices::FinalizeOpenCreditService, type: :service do
+  let(:service) { described_class.new(invoice:) }
+
+  let(:organization) { create(:organization, email_settings: Organization::EMAIL_SETTINGS) }
+  let(:invoice) { create(:invoice, organization:, invoice_type: 'credit', status: :open, payment_due_date: 1.week.ago.to_date) }
+
+  before do
+    allow(invoice).to receive(:should_sync_invoice?).and_return(true)
+    allow(invoice).to receive(:should_sync_sales_order?).and_return(true)
+  end
+
+  describe '.call' do
+    around { |test| lago_premium!(&test) }
```

Comment:
In this case, the premium was only added to test if the email was sent. I updated the test to remove the premium.

```rb
def should_deliver_email?
  License.premium? && invoice.organization.email_settings.include?('invoice.finalized')
end
```
I think the service should work even if you're not premium. I'm thinking if you created an open invoice because you were premuim but your license expire, the flow should continue and finalize the invoice. You won't be able to create new open invoices until the invoice is renewed.